api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +145,36 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	imagesBeforeLoad, err := driver.docker.ListImages(context.Background())
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	if conf.MaxImageCacheSize != 0 {
+		driver.imageCache = NewCache(int64(conf.MaxImageCacheSize))
+
+		go func() {
```

Comment:
Do we ever care about when this function completes?  Read: wondering if there's a need to add a channel to wait for it, in case we need to ensure the cache has been populated.  Just makes my race-condition-spidey-sense tingle a bit, but that could also be allergies.